xdsl/transforms/experimental/ConvertStencilToLLMLIR.py

```diff
@@ -288,55 +272,84 @@ def match_and_rewrite(self, op: AccessOp, rewriter: PatternRewriter, /):
         rewriter.replace_matched_op([*off_const_ops, *off_sum_ops, load], [load.res])
 
 
-@dataclass
 class StencilTypeConversionFuncOp(RewritePattern):
-    return_targets: dict[ReturnOp, list[CastOp | memref.Subview | None]]
-
     @op_type_rewrite_pattern
     def match_and_rewrite(self, op: FuncOp, rewriter: PatternRewriter, /):
         inputs: list[Attribute] = []
         for arg in op.body.block.args:
             if isa(arg.typ, FieldType[Attribute]):
-                memreftyp = GetMemRefFromField(arg.typ)
+                memreftyp = StencilToMemRefType(arg.typ)
                 rewriter.modify_block_argument_type(arg, memreftyp)
                 inputs.append(memreftyp)
             else:
                 inputs.append(arg.typ)
+        outputs: list[Attribute] = [
+            StencilToMemRefType(out) if isa(out, FieldType[Attribute]) else out
+            for out in op.function_type.outputs
+        ]
+        op.attributes["function_type"] = FunctionType.from_lists(inputs, outputs)
 
-        op.attributes["function_type"] = FunctionType.from_lists(
-            inputs, list(op.function_type.outputs.data)
-        )
 
+class TypeSpreadingForOp(RewritePattern):
```

Comment:
let's think of a better name

Reply:
ToBeReimplementedAsInfrastructure?

Reply:
(Note to myself) more importantly, comment.

Reply:
> UpdateCarriedLoopVarTypes?

I like this one.